cloudfixManager.go

```diff
@@ -95,19 +141,31 @@ func (c *CloudfixManager) createMap(reccos []byte, attrMapping []byte) map[strin
 	return mapping
 }
 
-func (c *CloudfixManager) parseReccos() map[string]map[string]string {
+func (c *CloudfixManager) getReccos() (map[string]map[string]string, *customError) {
 	//function to process the reccomendations from cloudfix and turn that into a map
 	//the structure of the map is resourceID -> Attribute type that needs to be targetted -> Ideal Attribute Value
 	// If there is no attribute that has to be targetted, attribute type would be filled with "NoAttributeMarker" and
 	//Attribute Value would be filled with any message that in the end has to be displayed to the user
-	currPWD, _ := exec.Command("pwd").Output()
-	currPWDStr := string(currPWD[:])
-	currPWDStrip := strings.Trim(currPWDStr, "\n")
-	currPWDStrip += "/reccos.json"
-	reccos, errR := ioutil.ReadFile(currPWDStrip)
-	if errR != nil {
-		//Add Error Log
-		panic(errR)
+
+	var cloudAuth CloudfixAuth
+	mapping := make(map[string]map[string]string)
+	token, errA := cloudAuth.getToken()
+	if errA != nil && errA.statusCode != STORAGE_ERROR {
+		return mapping, errA
+	}
+	// currPWD, _ := exec.Command("pwd").Output()
+	// currPWDStr := string(currPWD[:])
+	// currPWDStrip := strings.Trim(currPWDStr, "\n")
+	// currPWDStrip += "/reccos.json"
+	// reccos, errR := ioutil.ReadFile(currPWDStrip)
+	// if errR != nil {
+	// 	//Add Error Log
+	// 	panic(errR)
+	// }
+	reccos, errT := c.getReccosFromCloudfix(token)
+	if errT != nil {
+		fmt.Println(errT.message)
+		return mapping, errT
 	}
 	attrMapping := []byte(`{
```

Comment:
this has to move out into a separate file hosted and downloaded when the program starts. rt? 

In that case have a separate class that is respondisble for providing this object. Later only that one class will need to change to download instead of returning a hardcoded map.

This whole thing of hosting, and download is something Anurodh can take up.

Reply:
There is no need for a reccos.json file now. Earlier, the cloudfixManager was reading the reccos from that file and then parsing it to make the recommendations available further. Now the cloudfixManager is getting them from the API directly and sending it further for parsing.